small-talk/app/accountCreate/page.js

```diff
@@ -13,12 +15,14 @@ export default function RegisterPage() {
     //const history = useHistory();
     const [textInput, setTextInput] = useState(null);
     
-    const RegisterPage = () => {
+    const RegisterPatient = async () => {
       // if login sucessful
+      await handleRegister(username, password, firstName, lastName, dob);
```

Comment:
Seems like the default logic here would still redirect to login if registration is unable to happen. Maybe we could have an alert  and not redirect? Just depends on how you imagined the user experience for a backend error.

Reply:
I believe the way the logic works is that when the admin presses the submit button, it'll perform the checkUP function. In this function, it will perform the necessary checks to ensure that all fields are filled, username/password does not contain each other, the username is not already taken, and that the password follows the specific pattern. If all of these checks pass, then this function will call the RegisterPatient(), which is the code you are referring to. At this point we believe that registration would be a success, so we would proceed to in add the credentials into the database (the handleRegister() does this.), then perform the redirect. Ultimately, the RegisterPatient() function only gets called when the all of the checks pass so it should be good? Please let us know if there is any amiss.

Reply:
Yeah your edit covers what I was meaning. The other checks verify that the user and password are valid, but the handleRegister can fail if for whatever reason mongodb has an error saving the new user

Reply:
Thank you for the suggestion. We have implemented the logic change.